### PR TITLE
dogecoin: cleanup and fix darwin build

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Supercedes #114769, #121941.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/dogecoin/default.nix [hunk 1]
```diff
@@ -1,12 +1,14 @@
 { lib, stdenv , fetchFromGitHub
 , pkg-config, autoreconfHook
 , db5, openssl, boost, zlib, miniupnpc, libevent
-, protobuf, util-linux, qt4, qrencode
-, withGui }:
+, protobuf, qtbase ? null
+, wrapQtAppsHook ? null, qttools, qmake ? null, qrencode
```
Comment (cole-h) [inline pkgs/applications/blockchains/dogecoin/default.nix:5]: I believe qttools is only available as a part of `libsForQt5`.

Comment (veprbl) [reply pkgs/applications/blockchains/dogecoin/default.nix:5]: This is why `libsForQt5.callPackage` is used

Comment (cole-h) [reply pkgs/applications/blockchains/dogecoin/default.nix:5]: Ah, I didn't check that, my bad. Dunno why it's not being found then. I'll have to take another look later.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -29495,7 +29495,7 @@ with pkgs;
   digibyte = libsForQt514.callPackage ../applications/blockchains/digibyte { withGui = true; };
   digibyted = callPackage ../applications/blockchains/digibyte { withGui = false; };
 
-  dogecoin  = callPackage ../applications/blockchains/dogecoin {
+  dogecoin  = libsForQt5.callPackage ../applications/blockchains/dogecoin {
     boost = boost165;
     withGui = true;
   };
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix:29501]: There is `dogecoind` on the next line, it needs `libsForQt5.callPackage` as well

Comment (prusnak) [reply pkgs/top-level/all-packages.nix:29501]: `dogecoind` is just a daemon without GUI, so it shouldn't require this. It's the same situation as with the `bitcoin` and `bitcoind`

File: pkgs/applications/blockchains/dogecoin/default.nix [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+, withGui, withUpnp ? true, withUtils ? true, withWallet ? true
+, withZmq ? true, zeromq, util-linux ? null, Cocoa ? null }:
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/dogecoin/default.nix:7]:
```suggestion
, withZmq ? true, zeromq, util-linux, Cocoa }:
```
if those are null something is broken.

File: pkgs/applications/blockchains/dogecoin/default.nix [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 
 with lib;
 stdenv.mkDerivation rec {
-  name = "dogecoin" + (toString (optional (!withGui) "d")) + "-" + version;
+  pname = "dogecoin" + (optionalString (!withGui) "d");
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/dogecoin/default.nix:11]:
```suggestion
  pname = "dogecoin" + optionalString (!withGui) "d";
```

File: pkgs/applications/blockchains/dogecoin/default.nix (hunk 3 above)
Comment (siraben) [reply pkgs/applications/blockchains/dogecoin/default.nix:7]: Not required arguments on darwin

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (veprbl) [reply pkgs/top-level/all-packages.nix:29501]: Qt is not only a GUI library

**Pull request merged**